Updated TrackballControls to run `update()` automatically.

arodic opened the pull request:
Currently trackball example renders the scene permanently, even while nothing is changing in the scene. This can drain your laptop battery quickly if you leave the example tab open. Let's try to update the examples to be a little more energy efficient.

Mugen87 commented:
When using on demand rendering, the animation loop should always be removed.

arodic commented:
@Mugen87 TrackballControls uses inertia/damping so continuous updates are needed to determine if rendering is needed (delta > EPS). An alternative is to start/stop animation loop dynamically but that would make the example too complicated imo. This example is already wasting energy with continuous animation loop and this is a simple fix to skip the rendering at least.
 
I can remove stats or move update to render function so it only updates on render.

Mugen87 commented:
How about disabling damping?

arodic commented:
Personally, I prefer damping disabled for camera controls in general but it seems to be a popular feature in the threejs community.

I think we should provide examples on how to use this feature efficiently. If people want to use it, they should have examples on how to use it without the brute-force infinite animation loop solution.

Mugen87 commented:
I have the feeling the example is now confusing, especially for beginners. I leave the final decision up to @mrdoob.

Commit:
Updated TrackballControls to run `update()` automatically

arodic commented:
I changed this PR so `update()` is no longer needed when using TrackballControls. Fixed a bug where pointermove and pointerup listeners were added twice.

It also removes unnecessary animation loop from multiple examples.

arodic commented:
I'm gonna close this PR since its not getting any traction. Might give it another shot in the future.
